python/casmsmartphase/MNVMerge.py

```diff
@@ -0,0 +1,313 @@
+        for form_line in format_lines:
+            for i in range(1, self.longest_MNV + 1):
+                lines_to_add.append(self.generate_new_increment_header(form_line, i))
+
+        for new_head_line in lines_to_add:
+            writer_header.add_line(new_head_line)
+
+        return writer_header
+
+    def merge_snv_to_mnv(self, snv_list):
+        """
+        Merge snvs from list into a single variant and output to VCF
+        """
+        do_qual = 0
+        do_filter = 0
+        qual = 0
+        filter = []
+        if snv_list[0].QUAL:
+            do_qual = 1
+            print(
+                "Found a QUAL value, output will be a mean of all QUAL.",
+                file=sys.stderr,
+                end="\n",
+            )
+        else:
+            qual = snv_list[0].QUAL
+
+        if snv_list[0].FILTER:
+            do_filter = 1
+            print(
+                "Found a FILTER value, output will be all FILTERs encountered at all bases.",
+                file=sys.stderr,
+                end="\n",
+            )
+        else:
+            filter = snv_list[0].FILTER
+
+        chrom = snv_list[0].CHROM
+        pos = snv_list[0].POS
+        id = []  # list of SNV IDs?
+        ref = ""
+        alt_str = ""
+        info = {}
+        format = []
+        # Setup new call objects
+        calls_dict = {}
+
+        for n, var in enumerate(snv_list, start=1):
+            ref += str(var.REF)
+            alt_str += str(var.ALT[0].value)
+
+            # Add incremental format strings
+            for f in var.FORMAT:
+                format.append(f + f"_{n}")
+
+            # Add incremented info
+            for key, val in var.INFO.items():
+                info[key + f"_{n}"] = val
+
+            # Calls, should be a NORMAL and TUMOUR call and associated counts
+            for call in var.calls:
+                new_call = None
+
+                # Check for existing new_call object
+                data_to_add = dict()
+                sample_nom = call.sample
+
+                # Build new data from existing call data (add increment)
+                for k, v in call.data.items():
+                    k = k + f"_{n}"
+                    data_to_add[k] = v
+                if call.sample in calls_dict:
+                    existing_call = calls_dict[call.sample]
+                    exist_data = existing_call.data.copy()
+
+                    # Append new data to previously processed
+                    for k, v in data_to_add.items():
+                        exist_data[k] = v
+                    data_to_add = exist_data
+
+                new_call = vcfpy.Call(sample_nom, data_to_add)
+                calls_dict[call.sample] = new_call
+
+            # If we want to append filters
+            if do_filter:
+                filter.append(var.FILTER)
+            # If we want to append qualities to generate a mean
+
```

Comment:
Line separates comment from code it belongs to

Reply:
Suspect black moved this from the end of the previous line. Moved it to line above